src/kili/adapters/kili_api_gateway/label/annotation_to_json_response.py

```diff
@@ -389,36 +385,15 @@ def _video_object_detection_annotation_to_json_response(
     return json_resp
 
 
-def _interpolate_vertex(
-    vertex_initial_state: Vertice,
-    vertex_final_state: Vertice,
-    initial_state_frame_index: int,
-    final_state_frame_index: int,
-    at_frame: int,
-) -> Vertice:
-    """Generate vertex at some frame between two vertice states."""
-    nb_frames_in_between = final_state_frame_index - initial_state_frame_index
-
-    return Vertice(
-        x=vertex_initial_state["x"]
-        + (vertex_final_state["x"] - vertex_initial_state["x"])
-        * (at_frame - initial_state_frame_index)
-        / nb_frames_in_between,
-        y=vertex_initial_state["y"]
-        + (vertex_final_state["y"] - vertex_initial_state["y"])
-        * (at_frame - initial_state_frame_index)
-        / nb_frames_in_between,
-    )
-
-
 def _interpolate_object_(
     *,
-    object_initial_state: List[Vertice],
+    object_type: str,
+    object_initial_state: List[List[List[Vertice]]],
     initial_state_frame_index: int,
-    object_final_state: List[Vertice],
+    object_final_state: List[List[List[Vertice]]],
     final_state_frame_index: int,
     at_frame: int,
-) -> List[Vertice]:
+) -> List[List[List[Vertice]]]:
```

Comment:
For better readability:

- make the functions private
- organize the code like a table of content: first the top level, then the functions called by the top level, then apply the recursion.